### PR TITLE
fix(openai): Add trace context to client requests

### LINKED ISSUE #2291: 🐛 Bug Report: Trace-ID Injection Missing Between OpenAI and vLLM Spans
### Which component is this bug for?

OpenAI Instrumentation

### 📜 Description

This bug is related to #1983. Similar to what's described there, the `OpenAIInstrumentor` doesn't propagate the trace context of the exported span through the LLM request.

This fix will also benefit other frameworks built on top of the OpenAI client library, such as CrewAI.

### 👟 Reproduction steps

1. Run Jaeger as the trace collector
```
docker run --rm --name jaeger \
    -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
    -p 6831:6831/udp \
    -p 6832:6832/udp \
    -p 5778:5778 \
    -p 16686:16686 \
    -p 4317:4317 \
    -p 4318:4318 \
    -p 14250:14250 \
    -p 14268:14268 \
    -p 14269:14269 \
    -p 9411:9411 \
    jaegertracing/all-in-one:1.57
```

2. Run vLLM
```
export OTEL_SERVICE_NAME="vllm-server"
export OTEL_EXPORTER_OTLP_TRACES_INSECURE=true
vllm serve meta-llama/Llama-3.2-1B-Instruct --otlp-traces-endpoint "grpc://localhost:4317"
```

3. Run OpenAI client with `OpenAIInstrumentor`:

```
from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.resources import Resource
from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
from opentelemetry.trace import set_tracer_provider


trace_provider = TracerProvider(resource=Resource(attributes={"service.name": "OpenAI-client"}))
trace_provider.add_span_processor(BatchSpanProcessor(OTLPSpanExporter(endpoint="grpc://localhost:4317", insecure=True)))
trace_provider.add_span_processor(BatchSpanProcessor(ConsoleSpanExporter()))
set_tracer_provider(trace_provider)


from opentelemetry.instrumentation.openai import OpenAIInstrumentor
OpenAIInstrumentor().instrument()

from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:8000/v1",
    api_key="EMPTY",
)

chat_completion = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": "Say this is a test",
        }
    ],
    model="meta-llama/Llama-3.2-1B-Instruct",
)
print(chat_completion)
```

### 👍 Expected behavior

I should see in Jaeger WebUI one trace with 2 spans.

### 👎 Actual Behavior with Screenshots

I see 2 traces, each with a single span.

http://localhost:16686/

image

image

### 🤖 Python Version

_No response_

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ronensc**
This PR resolves #2291.

A new flag, `enable_trace_context_propagation,` has been introduced to control trace context propagation. By default, this flag is set to `true`, ensuring that the trace context is propagated.

- [X] I have added tests that cover my changes.
- [ ] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.
- [X] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [ ] (If applicable) I have updated the documentation accordingly.

### CODE REVIEW

File: packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/__init__.py [hunk 1]
```diff
@@ -23,13 +23,15 @@ def __init__(
...
         ] = lambda *args: "",
+        disable_trace_context_propagation: bool = False,
     ):
         super().__init__()
         Config.enrich_assistant = enrich_assistant
         Config.enrich_token_usage = enrich_token_usage
         Config.exception_logger = exception_logger
         Config.get_common_metrics_attributes = get_common_metrics_attributes
         Config.upload_base64_image = upload_base64_image
+        Config.disable_trace_context_propagation = disable_trace_context_propagation
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/__init__.py:34]: nit: I think it's better to have "positive flags" - so this should be "enable_trace_context_propagation" and default to `True`. WDYT?

**Review by nirga — COMMENTED** [review]
Thanks @ronensc! Great work :) 
Left a small comment

File: packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/__init__.py (hunk 1 above)
Comment (ronensc) [reply packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/__init__.py]: sure, no problem. done :)

**Review by nirga — APPROVED** [review]
Thanks @ronensc!

**Pull request merged**